packages/client/hooks/useMenu.ts

```diff
@@ -53,11 +63,30 @@ const useMenu = <T extends HTMLElement = HTMLButtonElement>(
     return Math.max(40, bbox ? bbox.width : 40)
   }, [originRef.current /* eslint-disable-line react-hooks/exhaustive-deps */])
   const {loadingDelay, loadingDelayRef} = useLoadingDelay()
+
+  const [topOffset, setTopOffset] = useState<number>(0)
+
+  const menuCoords = useMemo(() => {
+    if (
+      portalStatus === PortalStatus.Mounted ||
+      portalStatus === PortalStatus.Entering ||
+      portalStatus === PortalStatus.Entered
+    ) {
+      setTopOffset(cardContainerRef?.current?.scrollTop || 0)
+    }
+    return (coords as {right: number; top: number}).top
```

Comment:
-1 it's better to extract a type for coords firstly.

Reply:
-2 I think you can use `'key' in object`, ref: https://www.freecodecamp.org/news/how-to-check-if-an-object-has-a-key-in-javascript/